src/test/scala/com/microsoft/hyperspace/index/HybridScanSuite.scala

```diff
@@ -404,7 +405,13 @@ trait HybridScanSuite extends QueryTest with HyperspaceSuite {
       val baseQuery = joinQuery()
       val basePlan = baseQuery.queryExecution.optimizedPlan
 
-      withSQLConf("spark.sql.autoBroadcastJoinThreshold" -> "-1") {
+      // RemoveRedundantProjects rule causes HybridScanForIcebergTest to fail
+      // in Spark 3.1. Either a bug of Spark 3.1, or Iceberg needs to be
+      // updated. Either way, it will take some time to be fixed, so let's
+      // temporarily disable the rule here.
+      withSQLConf(
+        "spark.sql.autoBroadcastJoinThreshold" -> "-1",
+        "spark.sql.execution.removeRedundantProjects" -> "false") {
```

Comment:
I'll check who's responsible for the failure, Spark or Iceberg. Or it could be us.

Reply:
Could you share the exception (or failure reason)?

Reply:
You can find the details [here](https://issues.apache.org/jira/browse/SPARK-35287). Basically, it's `java.lang.ClassCastException: org.apache.spark.sql.catalyst.expressions.GenericInternalRow cannot be cast to org.apache.spark.sql.catalyst.expressions.UnsafeRow`. Seems unrelated to Hyperspace.

Reply:
Note: the bug will be fixed in Spark 3.2.0 and 3.1.3.